flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/StreamSourceOperatorWatermarksTest.java

```diff
@@ -147,7 +147,7 @@ public void testAutomaticWatermarkContext() throws Exception {
 		StreamSourceContexts.getSourceContext(
 			TimeCharacteristic.IngestionTime,
 			processingTimeService,
-			operator.getContainingTask().getCheckpointLock(),
+			((MockStreamTask<?, ?>) operator.getContainingTask()).getCheckpointLock(),
```

Comment:
It would be cleaner to return the task from `setupSourceOperator` instead of the task.